### PR TITLE
Add cron editor

### PR DESCRIPTION
**Pull request by PPiedel**

### CODE REVIEW

File: ui/client/components/graph/node-modal/editors/expression/Cron/CronEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import {ExpressionObj} from "../types"
+import {Validator} from "../../Validators"
+import React, {useEffect, useRef, useState} from "react"
+import Cron from 'react-cron-generator'
+import "react-cron-generator/dist/cron-builder.css"
+import Input from "../../field/Input"
+import "./cronEditorStyle.styl"
+import _ from "lodash"
```
Comment (lciolecki) [inline ui/client/components/graph/node-modal/editors/expression/Cron/CronEditor.tsx]: Please import only used method from lodash. In fact we should use favors from ts :)

File: ui/client/components/graph/node-modal/editors/expression/Editor.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ import {DateTimeEditor} from "./DateTimeEditor/DateTimeEditor"
 import {Error, Validator, validators, validatorType} from "../Validators"
 import DurationEditor from "./duration/DurationEditor"
 import PeriodEditor from "./duration/PeriodEditor"
+import CronEditor from "./Cron/CronEditor";
```
Comment (lciolecki) [inline ui/client/components/graph/node-modal/editors/expression/Editor.ts]: Semicolon

File: ui/client/components/graph/node-modal/editors/expression/Cron/CronEditor.tsx (hunk 1 above)
Comment (PPiedel) [reply ui/client/components/graph/node-modal/editors/expression/Cron/CronEditor.tsx]: Done, using ts

File: ui/client/components/graph/node-modal/editors/expression/Editor.ts (hunk 2 above)
Comment (PPiedel) [reply ui/client/components/graph/node-modal/editors/expression/Editor.ts]: Done

File: ui/client/components/graph/node-modal/editors/expression/Cron/CronEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    </div>
+  )
+}
+
+CronEditor.switchableTo = (expressionObj: ExpressionObj) =>
+  CRON_DECODE_REGEX.test(expressionObj.expression) || expressionObj.expression === ""
+
+CronEditor.switchableToHint = "Switch to basic mode"
+
+CronEditor.notSwitchableToHint = "Expression must match pattern T(com.cronutils.parser.CronParser).parse('* * * * * * *') to switch to basic mode"
```
[the remark this answers is not in the document]
Comment (PPiedel) [reply ui/client/components/graph/node-modal/editors/expression/Cron/CronEditor.tsx]: Done

**Pull request merged**